backend/owa.ts

```diff
@@ -1,7 +1,20 @@
 import { session as Session, BrowserWindow } from "electron";
 
+const kCookieName = "X-OWA-CANARY";
+const kHotmailServer = "outlook.live.com";
+
 export async function fetchSessionData(partition: string, url: string, interactive: boolean) {
   let session = Session.fromPartition(partition);
+  let urlObj = new URL(url);
+  let cookies = await session.cookies.get({ name: kCookieName });
+  if (cookies.length) {
+    urlObj.hostname = cookies[0].domain;
+    let owaCookies = cookies.filter(cookie => cookie.path == "/owa/");
```

Comment:
That won't work. It also needs to be `startsWith("/owa/")`

Reply:
Your logic is still different from Owl. Why don't you just use the code logic? That's known to work.

```
   onCookie() ...
      if (isHotmail &&
          details.cookie.domain == kHotmailServer &&
          this.cookieNames.includes(details.cookie.name) &&
          details.cookie.path.startsWith("/owa/")) {
        // Hotmail also sets cookies for /owa/0/, /mail/0/, /calendar/0/ etc.,
        // but we can use only the /owa/0/ cookie.
        // We also need to use that URL for the service request.
        // This needs to happen before CheckLoginFinished().
        this.serverURL.pathname = details.cookie.path;
      }
```

Please include the comment as well.